### DISCUSSION TITLE
Query by ref in papis

### DISCUSSION THREAD

**Discussion by awa5114** [papis/papis#933] [Q&A] [answered]
Can I query by document reference in papis? I have a document with the following info:

```
author: Li, Shuping and Yu, Weiming
booktitle: Conference on Environmental Pollution and Public Health
files:
- test.pdf
papis_id: 67ecdadd2045e006757516420a3d1d77
ref: Li_Shuping_and_Yu_Weiming
time-added: 2024-11-03-17:12:01
title: Urban Water Balance Analsis by Holistic Approach
type: inproceedings
year: 2010
```

But the following doesn't seem to work:

`papis list --file Urban_Water_Bal`

I also tried other variants like:

`papis list --file ref:Urban_Water_Bal`

`papis list --file "ref : Urban_Water_Bal"`

But those don't seem to work either

**Comment by alexfikl** [comment, marked as answer]
That should definitely be possible. What version of Papis are you on? What database backend is that?

> papis list --file Urban_Water_Bal

(Assuming the default `papis` backend) To have this work, you have to add `doc[ref]` to your `match-format` setting https://papis.readthedocs.io/en/latest/configuration.html#confval-match-format. By default it doesn't include the `ref`, so searching for it without a qualifier won't work.

> papis list --file ref:Urban_Water_Bal

Hm, this works for me. Maybe doing a `papis cache reset` (or `papis --clear-cache` on older versions) will help?

Also, I'm not sure if this is a typo but the ref you're looking for and the one in the file are different:
```
ref: Li_Shuping_and_Yu_Weiming
```

**Reply by jghauser** [reply]
`papis list --file ref:Urban_Water_Bal` works for me with the `whoosh` backend. (Just adding this cause I think @alexfikl is using the `papis` backend.)

EDIT: Though maybe you'd need to add `ref` to [`whoosh-schema-fields`](https://papis.readthedocs.io/en/latest/configuration.html#confval-whoosh-schema-fields). @alexfikl, am I right that all fields that should be searchable must be added here? Why does it only include `doi` by default?

**Reply by alexfikl** [reply]
Yeah, you'd need to add it to the `whoosh-schema-prototype` thing for it to be searchable
https://papis.readthedocs.io/en/latest/configuration.html#confval-whoosh-schema-prototype
The `whoosh-schema-fields` list is just a quicker way to add some `TEXT` fields in there, since the `-prototype` input is a bit unwieldy.
https://github.com/papis/papis/blob/f390fd79df82b9458ac5f909f5cde5d1b919c9b4/papis/database/whoosh.py#L299-L305

Adding the `ref` in there by default sounds like a good idea though. I think the `papis` backend also has the tags added by default, so we should definitely make them more consistent :(

**Comment by awa5114** [comment]
@alexfikl  Sorry, I was making a dumb mistake on my end. It definitely works but only with an adjusted `match-format` in the config file as you mentioned.

**Reply by alexfikl** [reply]
Glad to hear it works! :tada:
